src/main/java/org/openrewrite/java/spring/boot2/ConvertToSecurityDslVisitor.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot2;
+
+import org.openrewrite.Cursor;
+import org.openrewrite.Tree;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.marker.Markup;
+
+import java.util.*;
+
+public class ConvertToSecurityDslVisitor<P> extends JavaIsoVisitor<P> {
+
+    private static final String MSG_FLATTEN_CHAIN = "http-security-dsl-flatten-invocation-chain";
+
+    private static final String MSG_TOP_INVOCATION = "top-method-invocation";
+
+    private String securityFqn;
+
+    private Collection<String> convertableMethods;
+
+    public ConvertToSecurityDslVisitor(String securityFqn, Collection<String> convertableMethods) {
+        this.securityFqn = securityFqn;
+        this.convertableMethods = convertableMethods;
+    }
+
+    @Override
+    public Space visitSpace(Space space, Space.Location loc, P executionContext) {
+        return space;
+    }
+
+    @Override
+    public Markers visitMarkers(Markers markers, P executionContext) {
+        return markers;
+    }
```

Comment:
```suggestion
```

Unless I'm missing something this looks unneeded. Perhaps code which was used as a convenient place to hang a breakpoint during debugging?

Reply:
I have removed those 2 methods.

However, I should probably explain why I add these...
I add those two methods on the IDE side for visitors that mark CUs for various issues. Those visitrs meant to be very quick hence I override visit space and markers to prune these AST branches as what I'm looking for isn't there. It just seem to win a a few ms for the visitor... Looks like space can contain comments, some of these comments are javadoc comments... also provenance-kind markers seem to be attached to all AST nodes (or many if not all and pruning markers seem to to help

Reply:
It makes sense that skipping parsing javadocs would save time. I wonder if we should make visiting javadoc comments opt-in for visitors, as most aren't going to be interested. Not sure what you mean about the markers, though. Most tree elements have no markers whatsoever. It is only the root element which has provenance markers on it.

Reply:
@sambsnyd it is the positions markers not provenance. Those markers are everywhere for me as I care about line/column positional data...